website/addons/base/views.py

```diff
@@ -516,21 +570,21 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
         raise HTTPError(httplib.BAD_REQUEST)
 
     if not isinstance(node_addon, StorageAddonBase):
-        raise HTTPError(httplib.BAD_REQUEST, {
+        raise HTTPError(httplib.BAD_REQUEST, data={
             'message_short': 'Bad Request',
-            'message_long': 'The add-on containing this file is no longer connected to the {}.'.format(node.project_or_component)
+            'message_long': 'The {} add-on containing {} is no longer connected to {}.'.format(provider, path, node.project_or_component)
         })
 
     if not node_addon.has_auth:
-        raise HTTPError(httplib.UNAUTHORIZED, {
+        raise HTTPError(httplib.UNAUTHORIZED, data={
             'message_short': 'Unauthorized',
-            'message_long': 'The add-on containing this file is no longer authorized.'
+            'message_long': 'The {} add-on containing {} is no longer authorized.'.format(provider, path)
         })
 
     if not node_addon.complete:
-        raise HTTPError(httplib.BAD_REQUEST, {
+        raise HTTPError(httplib.BAD_REQUEST, data={
             'message_short': 'Bad Request',
-            'message_long': 'The add-on containing this file is no longer configured.'
+            'message_long': 'The {} add-on containing {} is no longer configured.'.format(provider, path)
```

Comment:
HTTP error messages (error.mako) are treated as trusted text.

The path (really, any user entered text inserted via `string.format` in HTTPErrors) will also need to be escaped with `markupsafe.escape` accordingly.